src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private int[] array;
+    private int n;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+        this.n = array.length;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length == 1) {
+            return;
```

Comment:
The base case check for the recursion should be when the length of the array is less than or equal to 1, as arrays of length 1 are already sorted by definition.